Python/bytecodes.c

```diff
@@ -582,9 +582,10 @@ dummy_func(
             PyObject *right_o = PyStackRef_AsPyObjectBorrow(right);
             assert(PyLong_CheckExact(left_o));
             assert(PyLong_CheckExact(right_o));
+            DEOPT_IF(!_PyLong_BothAreCompact((PyLongObject *)left_o, (PyLongObject *)right_o));
```

Comment:
If you narrow the checks in `_GUARD_TOS_INT` and `_GUARD_NOS_INT` to check for compact ints, we can drop the check here. That way we keep the guard and action in separate micro-ops.

Likewise for `+` and `-`.   The only complexity is that `optimizer_bytecodes.c` will need updating to handle compact ints differently.

This change is fine for now though, if you want to do the more sophisticated change in another PR.

Reply:
Sure, I can do that.